src/sif/truckcost.cc

```diff
@@ -368,6 +372,12 @@ TruckCost::TruckCost(const Costing& costing)
   for (uint32_t d = 0; d < 16; d++) {
     density_factor_[d] = 0.85f + (d * 0.025f);
   }
+
+  // determine what to do with hgv=no edges
+  bool no_hgv_access_penalty_active = !(costing_options.hgv_no_access_penalty() == kMaxPenalty);
+  no_hgv_access_penalty_ = no_hgv_access_penalty_active * costing_options.hgv_no_access_penalty();
```

Comment:
nit: is this necessary, cant you just set it regardless? it will only be applied when the access doesnt have truck right? which wont happen anyway if its not "active" or am i missing something? just a nit not important

Reply:
if we don't do it here and want to use it in transitioncost, then we'd have to check the same thing, i.e. if it's ` == kMaxPenalty`. instead of having that ternary being done every time we call TransitionCost, I'd rather do it once here.